### DISCUSSION TITLE
Title sort order reversed

### DISCUSSION THREAD

**Discussion by onlinemumie** [standardnotes/forum#2800] [Suggestions]
The last day or so, I've noted that the web app is acting up:
1. Title sort order is reversed when set to normal.
2. Many smart tags don't work anymore.

It's the same on both Linux and Windows - in Brave, Firefox, and Google Chrome.

The desktop and android apps behave as normal.

**Comment by atmoio** [comment]
Thanks for reporting. The web app is a version ahead of the desktop app. I do see that title sort is reversed. As for smart views, could you give me examples of views that are failing for you, if you can recall or replicate their syntax?

**Comment by onlinemumie** [comment]
Yes, here are a couple of ones that _don't_ work:

- - -

!["Arbete ! &", "ignored", "and", [["tags", "includes", ["title", "startsWith", "Arbete"]], ["title", "matches", "^(!|&)"]]]

!["Arbete .=Deadline", "ignored", "and", [["tags", "includes", ["title", "startsWith", "Arbete"]], ["title", "matches", "^(?!.*(R\\. Smarta taggar|R\\. Uppmaningar))"], ["text", "matches", "\\.=Deadline"]]]

- - -

And here are a copule ones that work (but with reversed titles):

!["Alla !", "title", "startsWith", "!"]

!["Alla .#Logg", "ignored", "and", [["title", "matches", "^(?!.*(R\\. Smarta taggar|R\\. Privat\\/Planering))"], ["text", "matches", "\\.\\#Logg"]]]

**Comment by atmoio** [comment]
Have confirmed a fix for the smart view issue, will ship out to web later today hopefully. Looking into sorting..

**Comment by atmoio** [comment]
Hmm so actually the latest version _fixes_ title sort order, which was always incorrect technically, but correct for humans. That is, the default sort for created date and modified date is **descending**—you want the most recent first (highest value), and then it descends to the lowest value. Title sort also uses descending, which means it starts at the highest value first (z), and goes down.

Right now the sort direction (descending or ascending) is global and not particular to which criteria you've chosen to sort by. So I'm not quite sure whether to change the code to make title sort technically wrong but perceptively correct, or to leave it as is 🤔

**Comment by onlinemumie** [comment]
I can see the logic in a descending sort order, at least when it comes to dates. On the other hand, in most file managers, the default sort order is ascending, probably because file name is the primary sort criteria. Perhaps this is a question of what kind of tool Standard Notes is trying to be. A file manager? A blog post system? A personal encyclopedia? Or something else entirely.

Aside from that, I don't think that the question of default sort order is that important as long as it is the same in the web, desktop and android versions. And I guess it will be as soon as the new version is out on all platforms.
